File: backend/app/modules/career/schemas/analysis.py
"""
=========================================================

SkillBattle

Analysis Schemas

=========================================================
"""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class AnalysisStatusResponse(BaseModel):

    resume_id: str

    status: str

    progress: int

    parsed: bool

    ai_processed: bool

    ats_score: int

    placement_score: int

    updated_at: datetime | None = None


class AnalysisResultResponse(BaseModel):

    resume_id: str

    status: str

    analysis: dict

File: src/types/analysis.ts
export interface AnalysisStatus {

    resume_id: string;

    status: string;

    progress: number;

    parsed: boolean;

    ai_processed: boolean;

    ats_score: number;

    placement_score: number;

}

export interface ResumeAnalysis {

    summary: string;

    strengths: string[];

    weaknesses: string[];

    missing_skills: string[];

    improvement_suggestions: string[];

    resume_score: number;

}